services/user-svc/internal/organization/organization.go

```diff
@@ -487,16 +517,18 @@ func GetOrganizationById(db *gorm.DB, id uuid.UUID) (*Organization, error) {
 	return &organization, nil
 }
 
-func GetInvitationById(db *gorm.DB, id uuid.UUID) (*Invitation, error) {
-	invite := Invitation{
-		ID: id,
-	}
+func GetInvitationByIdAndEmail(db *gorm.DB, email string, id uuid.UUID) (*Invitation, error) {
 
-	if err := db.First(&invite).Error; err != nil {
-		return nil, err
+	var invitation Invitation
+	if err := db.Where("id = ? AND email = ?", id, email).First(&invitation).Error; err != nil {
+		if hwgorm.IsOurFault(err) {
+			return nil, status.Error(codes.Internal, err.Error())
+		} else {
+			return nil, status.Error(codes.InvalidArgument, fmt.Sprintf("invitation with id %s and email %s not found", id, email))
+		}
```

Comment:
Just return the gorm error. The gRPC errors should be handled on the method level above.